pkg/cmd/openshift-kube-apiserver/openshiftkubeapiserver/patch.go

```diff
@@ -30,72 +34,103 @@ import (
 		postStartHooks: map[string]genericapiserver.PostStartHookFunc{},
 	}
-	return func(config *master.Config, internalInformers internalinformers.SharedInformerFactory, kubeInformers clientgoinformers.SharedInformerFactory, pluginInitializers *[]admission.PluginInitializer, stopCh <-chan struct{}) (genericapiserver.DelegationTarget, error) {
-		kubeAPIServerInformers, err := NewInformers(internalInformers, kubeInformers, config.GenericConfig.LoopbackClientConfig)
+	return func(genericConfig *genericapiserver.Config, internalInformers internalinformers.SharedInformerFactory, kubeInformers clientgoinformers.SharedInformerFactory, pluginInitializers *[]admission.PluginInitializer) (genericapiserver.DelegationTarget, error) {
+		kubeAPIServerInformers, err := NewInformers(internalInformers, kubeInformers, genericConfig.LoopbackClientConfig)
 		if err != nil {
 			return nil, err
 		}
 
 		// AUTHENTICATOR
 		authenticator, postStartHooks, err := NewAuthenticator(
 			*kubeAPIServerConfig,
-			config.GenericConfig.LoopbackClientConfig,
+			genericConfig.LoopbackClientConfig,
 			kubeAPIServerInformers.OpenshiftOAuthInformers.Oauth().V1().OAuthClients().Lister(),
 			kubeAPIServerInformers.OpenshiftUserInformers.User().V1().Groups())
 		if err != nil {
 			return nil, err
 		}
-		config.GenericConfig.Authentication.Authenticator = authenticator
+		genericConfig.Authentication.Authenticator = authenticator
 		for key, fn := range postStartHooks {
 			patchContext.postStartHooks[key] = fn
 		}
 		// END AUTHENTICATOR
 
 		// AUTHORIZER
+		genericConfig.RequestInfoResolver = configprocessing.OpenshiftRequestInfoResolver()
 		authorizer := NewAuthorizer(internalInformers, kubeInformers)
-		config.GenericConfig.Authorization.Authorizer = authorizer
+		genericConfig.Authorization.Authorizer = authorizer
 		// END AUTHORIZER
 
 		// ADMISSION
-		projectCache, err := openshiftapiserver.NewProjectCache(kubeAPIServerInformers.InternalKubernetesInformers.Core().InternalVersion().Namespaces(), config.GenericConfig.LoopbackClientConfig, kubeAPIServerConfig.ProjectConfig.DefaultNodeSelector)
+		projectCache, err := openshiftapiserver.NewProjectCache(kubeAPIServerInformers.InternalKubernetesInformers.Core().InternalVersion().Namespaces(), genericConfig.LoopbackClientConfig, kubeAPIServerConfig.ProjectConfig.DefaultNodeSelector)
 		if err != nil {
 			return nil, err
 		}
 		clusterQuotaMappingController := openshiftapiserver.NewClusterQuotaMappingController(kubeAPIServerInformers.InternalKubernetesInformers.Core().InternalVersion().Namespaces(), kubeAPIServerInformers.InternalOpenshiftQuotaInformers.Quota().InternalVersion().ClusterResourceQuotas())
-		kubeClient, err := kubernetes.NewForConfig(config.GenericConfig.LoopbackClientConfig)
+		patchContext.postStartHooks["quota.openshift.io-clusterquotamapping"] = func(context genericapiserver.PostStartHookContext) error {
+			go clusterQuotaMappingController.Run(5, context.StopCh)
```

Comment:
any deeper reason why this is now a post start hook and not a plugin initializer?

Reply:
> any deeper reason why this is now a post start hook and not a plugin initializer?

There's a controller piece that neesd to run